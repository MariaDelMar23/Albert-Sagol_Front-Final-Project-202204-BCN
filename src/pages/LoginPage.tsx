import Header from "../components/Header/Header";
import LoginForm from "../components/LoginForm/LoginForm";

const LoginPage = () => {
  return (
    <>
      <Header pageName={"Login"} adminUser={false} />
      <LoginForm />
    </>
  );
};

export default LoginPage;
