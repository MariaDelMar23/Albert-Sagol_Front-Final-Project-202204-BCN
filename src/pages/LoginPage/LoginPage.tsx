import Header from "../../components/Header/Header";
import LoginForm from "../../components/LoginForm/LoginForm";

const LoginPage = (): JSX.Element => {
  return (
    <>
      <Header pageName={"Login"} adminUser={false} />
      <LoginForm />
    </>
  );
};

export default LoginPage;
