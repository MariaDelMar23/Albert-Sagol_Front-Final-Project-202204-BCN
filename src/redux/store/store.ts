import { configureStore } from "@reduxjs/toolkit";
import petsReducer from "../features/petsSlice";
import userReducer from "../features/userSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    pets: petsReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
