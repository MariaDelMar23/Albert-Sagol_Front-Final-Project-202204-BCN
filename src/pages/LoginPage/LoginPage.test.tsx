import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../../redux/store/store";
import LoginPage from "./LoginPage";
import TestRenderer from "react-test-renderer";

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("Given the LoginPage page", () => {
  describe("When it's called", () => {
    test("Then it should render a heading with the text 'Login'", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <LoginPage />
          </Provider>
        </BrowserRouter>
      );

      expect(
        screen.getByRole("heading", { level: 1, name: "Login" })
      ).toBeInTheDocument();
    });

    test("Then it should render a label with 'Username'", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <LoginPage />
          </Provider>
        </BrowserRouter>
      );

      expect(screen.getByLabelText("Username")).toBeInTheDocument();
    });
  });

  describe("When it's called and there are a token on the local Storage", () => {
    test("Then it should call navigate", () => {
      Object.defineProperty(window, "localStorage", {
        value: { getItem: jest.fn(() => "token") },
      });

      render(
        <BrowserRouter>
          <Provider store={store}>
            <LoginPage />
          </Provider>
        </BrowserRouter>
      );
      expect(mockNavigate).toHaveBeenCalled();
    });

    test("Then it should match with the snapshot", () => {
      const loginPageSnapshot = TestRenderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <LoginPage />
          </Provider>
        </BrowserRouter>
      );

      expect(loginPageSnapshot).toMatchSnapshot();
    });
  });
});
