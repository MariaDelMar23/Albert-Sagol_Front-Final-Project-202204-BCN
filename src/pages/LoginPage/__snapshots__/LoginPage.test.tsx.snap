// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the LoginPage page When it's called and there are a token on the local Storage Then it should match with the snapshot 1`] = `
Array [
  <header
    className="sc-bczRLJ QbMXw"
  >
    <div
      className="container--header-logo-title"
    >
      <img
        alt="Pet's House logo"
        height={74}
        src="./images/logo.svg"
        width={152}
      />
    </div>
    <h1>
      Login
    </h1>
    <div
      className="container--header-buttons"
    />
  </header>,
  <form
    autoComplete="off"
    className="sc-gsnTZi brwyZO login-form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="username"
      >
        Username
        <input
          id="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </label>
      <label
        htmlFor="password"
      >
        Password
        <input
          id="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </label>
      <button
        disabled={true}
        type="submit"
      >
        Login
      </button>
    </div>
    <div>
      <p>
        If you want to create an account
      </p>
      <a
        href="/register"
        onClick={[Function]}
      >
        Create account
      </a>
    </div>
  </form>,
]
`;
